app/controllers/templates_controller.rb

```diff
@@ -79,13 +79,19 @@ def auto_complete_controller_name
   end
 
   def preview
-    base = @template.preview_host_collection
-    @host = params[:preview_host_id].present? ? base.find(params[:preview_host_id]) : base.first
+    if params[:id]
+      find_resource
+      base = @template.preview_host_collection
+      @host = params[:preview_host_id].present? ? base.find(params[:preview_host_id]) : base.first
+    else
+      type = request.original_url.split("/")[4]
```

Comment:
this won't work correctly for users installing foreman on suburi (/foreman/), is there a better way to determine type? why not using `params[type_name_singular]`?